commit ca18c7e9bd20ba26eae91ef80617d1ee5527d193
Author: Dmitry Antipov <dmanti@microsoft.com>
Date:   Wed Nov 28 01:56:09 2018 +0000

    Merged PR 508: Resize some PCDs in EmbeddedPkg to UINT64 to fit addresses above 4G
    
    Resize some PCDs in EmbeddedPkg to UINT64 to fit addresses above 4G

diff --git a/EmbeddedPkg/EmbeddedPkg.dec b/EmbeddedPkg/EmbeddedPkg.dec
index 69922802f4..33f57c2f09 100644
--- a/EmbeddedPkg/EmbeddedPkg.dec
+++ b/EmbeddedPkg/EmbeddedPkg.dec
@@ -91,15 +91,18 @@
   gEmbeddedTokenSpaceGuid.PcdSataSiI3132FeatureDirectCommandIssuing|FALSE|BOOLEAN|0x00000051
 
 [PcdsFixedAtBuild.common]
-  gEmbeddedTokenSpaceGuid.PcdPrePiStackBase|0|UINT32|0x0000000b
-  gEmbeddedTokenSpaceGuid.PcdPrePiStackSize|131072|UINT32|0x0000000c
+  # MS_CHANGE START
+  # Change the PCDs to UINT64s to fit addresses above 4G
+  gEmbeddedTokenSpaceGuid.PcdPrePiStackBase|0|UINT64|0x0000000b
+  gEmbeddedTokenSpaceGuid.PcdPrePiStackSize|131072|UINT64|0x0000000c
 
-  gEmbeddedTokenSpaceGuid.PcdMemoryBase|0x0|UINT32|0x0000004e
-  gEmbeddedTokenSpaceGuid.PcdMemorySize|0x0|UINT32|0x0000004f
+  gEmbeddedTokenSpaceGuid.PcdMemoryBase|0x0|UINT64|0x0000004e
+  gEmbeddedTokenSpaceGuid.PcdMemorySize|0x0|UINT64|0x0000004f
 
-  gEmbeddedTokenSpaceGuid.PcdFlashFvMainBase|0x0|UINT32|0x00000043
+  gEmbeddedTokenSpaceGuid.PcdFlashFvMainBase|0x0|UINT64|0x00000043
   gEmbeddedTokenSpaceGuid.PcdFlashFvMainOffset|0x0|UINT32|0x00000044
-  gEmbeddedTokenSpaceGuid.PcdFlashFvMainSize|0x0|UINT32|0x00000045
+  gEmbeddedTokenSpaceGuid.PcdFlashFvMainSize|0x0|UINT64|0x00000045
+  # MS_CHANGE END
 
 # Used to help reduce fragmentation in the EFI memory map
 # EFI Pages (4K) are the units used
@@ -120,8 +123,11 @@
 
   gEmbeddedTokenSpaceGuid.PcdMetronomeTickPeriod|100|UINT32|0x00000020
 
-  gEmbeddedTokenSpaceGuid.PcdEmbeddedFdBaseAddress|0xffff0000|UINT32|0x00000030
-  gEmbeddedTokenSpaceGuid.PcdEmbeddedFdSize|0x0000000|UINT32|0x00000031
+  # MS_CHANGE START
+  # Change the PCDs to UINT64s to fit addresses above 4G
+  gEmbeddedTokenSpaceGuid.PcdEmbeddedFdBaseAddress|0xffff0000|UINT64|0x00000030
+  gEmbeddedTokenSpaceGuid.PcdEmbeddedFdSize|0x0000000|UINT64|0x00000031
+  # MS_CHANGE END
   gEmbeddedTokenSpaceGuid.PcdEmbeddedPerformanceCounterFrequencyInHz|0x0000000|UINT64|0x00000032
   gEmbeddedTokenSpaceGuid.PcdEmbeddedPerformanceCounterPeriodInNanoseconds|0x0000000|UINT32|0x00000033
 
